build.xml

```diff
@@ -240,11 +245,81 @@
         <string>--add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED</string>
 
     </resources>
-    <pathconvert property="_jvm_args_concat" refid="_jvm11_arg_items" pathsep=" "/>
-    <condition property="java11-jvmargs" value="${_jvm_args_concat}" else="">
-        <not>
-            <equals arg1="${ant.java.version}" arg2="1.8"/>
-        </not>
+    <pathconvert property="_jvm11_args_concat" refid="_jvm11_arg_items" pathsep=" "/>
+    <condition property="java-jvmargs" value="${_jvm11_args_concat}">
+        <equals arg1="${ant.java.version}" arg2="11"/>
+    </condition>
+
+    <resources id="_jvm22_arg_items">
+        <string>-XX:+UnlockDiagnosticVMOptions</string>
+        <string>-Djdk.attach.allowAttachSelf=true</string>
+        <string>-XX:+UseG1GC</string>
+        <string>-XX:+ParallelRefProcEnabled</string>
+        <!-- Below two are used in Cassandra, not in CNDB-->
+        <!--<string>-XX:MaxTenuringThreshold=1</string>
+        <string>-XX:G1HeapRegionSize=16m</string>-->
+        <string>-XX:G1RSetUpdatingPauseTimePercent=5</string>
+        <string>-XX:MaxGCPauseMillis=100</string>
+
+        <!-- Copy-paste for now these options from cndb/docker-entrypoint.sh; we will verify later whether we
```

Comment:
This comment should be removed when things are settled

Reply:
How do these all work?  You mention these are for tests in the comment, why are there both _jvm22_arg_items and _jvm22_test_arg_items?

It would be nice if we did not need to have all these duplicated here and also in the server.options files, but that is probably something for another ticket.

Reply:
> It would be nice if we did not need to have all these duplicated here and also in the server.options files, but that is probably something for another ticket.

Yes and no. Good reference: [CASSANDRA-18263](https://issues.apache.org/jira/browse/CASSANDRA-18263)

So pre-CMS being out - Apache Cassandra was using CMS for testing, but a different config for prod. Optimization for our tests or so. When we moved past JDK8 - CASSANDRA-18263 was born. Until someone determines what we do in tests, we will keep the same configuration as the server.options files for now. So far no one has reported any issues around that, and I am not sure we will get to differ anything again. Exactly the opposite, @jacek-lewandowski did very good points in favor to keep G1GC and the same config as prod.
But there is one more caveat - sometimes we need to open/export some JDK internals just for some of the test frameworks we use, but not for prod code. In those cases we will want that to be done only in build.xml. 

TLDR - let's consider a change in a different ticket.

Reply:
I believe [CASSANDRA-18263](https://issues.apache.org/jira/browse/CASSANDRA-18263) can be used for that and there is no need of additional one. I will leave a comment there that we may want to reconsider how we stream options between server.options and build.xml to reduce duplications and minimize the potential for bugs.

Reply:
Actually, I think in this part we won't be sharing much with OSS because we will likely be ahead all the time with JVM support. 

Therefore, I suggest doing the following:
1. Sets of exports and opens of the 11 are included in the corresponding sets of the 22, leverage that fact just having j11 args with them and j22 args with extra stuff j22 needs; then add both to the command line
2. Don't use different GC settings, we can share that as well

This way we can avoid duplications and problems

Reply:
I disagree here. I want to be able to see what is needed for which JDK and have things split. We also remove some options in newer JDKs or add new ones, and we want to mimic what we use in tests to what we use in the jvm options files. 
Example:
We need `<string>--add-opens java.base/jdk.internal.util.jar=ALL-UNNAMED</string>` only with JDK11.
By not including it in the tests for JDK17, I saw in tests I am still getting warnings that it is not needed and it triggered me to find out that we do not read the right jvm options file in Python dtests (which mimic real world where we use those files and not build.xml). And this was after I removed the option from the file I was supposed to use with newer JDK.

Not having explicit config per JDKs is also risky for people not to realize what they test or what they are supposed to test.